extensions/sql-database-projects/src/models/dataSources/sqlConnectionStringSource.ts

```diff
@@ -100,7 +112,11 @@ export class SqlConnectionDataSource extends DataSource {
 			providerName: 'MSSQL',
 			saveProfile: true,
 			id: this.name + '-dataSource',
-			options: []
+			options: {
+				'encrypt': this.encrypt,
```

Comment:
it might be possible to iterate through the connection string and build this dictionary in a more generic way, allowing you to send all options without special-casing these three.  Though I tend to agree sending the connection string to the connection API and having that code there is probably a better design.

Reply:
So the dictionary is getting created for the whole connection string here: https://github.com/microsoft/azuredatastudio/blob/f45069e4a99fce0e25a5fb773e1a555940fd0a4c/extensions/sql-database-projects/src/models/dataSources/sqlConnectionStringSource.ts#L83
But agreed it is not generic and while creating the connectionProfile object is when it actually misses the options. Hopefully with the new API it should work/generate connectionProfile as expected.